core/src/test/java/io/confluent/kafka/schemaregistry/SASLClusterTestHarness.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright 2016 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafka.schemaregistry;
+
+import kafka.security.minikdc.MiniKdc;
+import kafka.server.KafkaConfig;
+import kafka.utils.JaasTestUtils;
+import kafka.utils.TestUtils;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.protocol.SecurityProtocol;
+import org.apache.kafka.common.security.authenticator.LoginManager;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Option;
+import scala.collection.Seq;
+
+import javax.security.auth.login.Configuration;
+import java.io.File;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Properties;
+
+// sets up SASL for ZooKeeper and Kafka.
+public class SASLClusterTestHarness extends ClusterTestHarness {
+  public static final String JAAS_CONF = "java.security.auth.login.config";
+  public static final String KRB5_CONF = "java.security.krb5.conf";
+  public static final String ZK_AUTH_PROVIDER = "zookeeper.authProvider.1";
+
+  private static MiniKdc kdc;
+  private static File kdcHome;
+  private static String krb5ConfPath;
+  private static File jaasFile;
+
+  private static final Logger log = LoggerFactory.getLogger(SASLClusterTestHarness.class);
+
+  public SASLClusterTestHarness() {
+    super(DEFAULT_NUM_BROKERS);
+  }
+
+  @Override
+  protected SecurityProtocol getSecurityProtocol() {
+    return SecurityProtocol.SASL_PLAINTEXT;
+  }
+
+  @BeforeClass
+  public static void setUpKdc() throws Exception {
+    destroySaslHelper();
```

Comment:
`destroySaslHelper` needs to be called in `setUpKdc`. The exact reason is unknown. My assumption is that Java's system properties get mucked up in some way that is not compatible with the MiniKdc and associated system properties. Kafka's tests do the same thing.